reframe/core/namespaces.py

```diff
@@ -59,7 +59,7 @@ def __repr__(self):
 
     def _raise_namespace_clash(self, name):
         '''Raise an error if there is a namespace clash.'''
-        raise ValueError(
+        raise KeyError(
             f'{name!r} is already present in the current namespace'
```

Comment:
Yes, I was thinking though, that perhaps it makes sense to leave that one as a `KeyError` since this class just implements a base namespace and the users are never exposed to this low level namespace mechanisms. Thoughts?

Reply:
Agreed. This is not the final exception thrown to the user. Let's leave it as such.